graphite/graphite.go

```diff
@@ -79,6 +82,13 @@ func (f *GraphitePublisher) Publish(metrics []plugin.Metric, cfg plugin.Config)
 		}
 		data := fmt.Sprintf("%v", m.Data)
 		logger.Debug("Metric ready to send %s:%s", key, data)
+
+		if strings.ContainsAny(key, illegal) {
+			log.Info("Original string is %s", key)
+			key = strings.Replace(key, illegal, "_", len(key))
```

Comment:
this way to replace illegal chars. It won't work. I'll share how to make it work with you.